src/main/java/nl/knaw/dans/bridge/plugin/lib/common/IAction.java

```diff
@@ -0,0 +1,28 @@
+package nl.knaw.dans.bridge.plugin.lib.common;
+
+import nl.knaw.dans.bridge.plugin.lib.exception.BridgeException;
+
+import java.io.File;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Class IAction
+ * Created by Eko Indarto.
+ * Some discussion about using Optional in parameters:
+ * https://stackoverflow.com/questions/31922866/why-should-java-8s-optional-not-be-used-in-arguments
+ * https://softwareengineering.stackexchange.com/questions/238954/is-there-any-reason-not-to-use-optional-as-a-method-argument-in-the-case-where-y
+ * I have chosen to use Optional in parameter.
+ */
```

Comment:
Shortly describe how this 'Action' is used, first transformation, then use that information to construct the 'deposit data' or 'bag' and finally deposit it. The resulting ResponsData is used to track the progress.... something like that.